plexapi/sync.py

```diff
@@ -1,42 +1,103 @@
 # -*- coding: utf-8 -*-
 import requests
-from plexapi import utils
+import plexapi
 from plexapi.exceptions import NotFound
+from plexapi.base import PlexObject, Playable
 
 
-class SyncItem(object):
-    """ Sync Item. This doesn't current work. """
-    def __init__(self, device, data, servers=None):
-        self._device = device
-        self._servers = servers
-        self._loadData(data)
+def init(replace_provides=False):
+    if replace_provides or not plexapi.X_PLEX_PROVIDES:
+        plexapi.X_PLEX_PROVIDES = 'sync-target'
+    else:
+        plexapi.X_PLEX_PROVIDES += ',sync-target'
+
+    plexapi.BASE_HEADERS['X-Plex-Sync-Version'] = '2'
+    plexapi.BASE_HEADERS['X-Plex-Provides'] = plexapi.X_PLEX_PROVIDES
+
+    # mimic iPhone SE
+    plexapi.X_PLEX_PLATFORM = 'iOS'
+    plexapi.X_PLEX_PLATFORM_VERSION = '11.4.1'
+    plexapi.X_PLEX_DEVICE = 'iPhone'
+
+    plexapi.BASE_HEADERS['X-Plex-Platform'] = plexapi.X_PLEX_PLATFORM
+    plexapi.BASE_HEADERS['X-Plex-Platform-Version'] = plexapi.X_PLEX_PLATFORM_VERSION
+    plexapi.BASE_HEADERS['X-Plex-Device'] = plexapi.X_PLEX_DEVICE
+    plexapi.BASE_HEADERS['X-Plex-Model'] = '8,4'
+    plexapi.BASE_HEADERS['X-Plex-Vendor'] = 'Apple'
+
+
+class SyncItem(PlexObject):
+    TAG = 'SyncItem'
+
+    def __init__(self, server, data, initpath=None, clientIdentifier=None):
+        super().__init__(server, data, initpath)
+        self.clientIdentifier = clientIdentifier
 
     def _loadData(self, data):
         self._data = data
-        self.id = utils.cast(int, data.attrib.get('id'))
-        self.version = utils.cast(int, data.attrib.get('version'))
+        self.id = plexapi.utils.cast(int, data.attrib.get('id'))
+        self.version = plexapi.utils.cast(int, data.attrib.get('version'))
         self.rootTitle = data.attrib.get('rootTitle')
         self.title = data.attrib.get('title')
         self.metadataType = data.attrib.get('metadataType')
         self.machineIdentifier = data.find('Server').get('machineIdentifier')
-        self.status = data.find('Status').attrib.copy()
+        self.status = Status(self._server, data.find('Status'))
         self.MediaSettings = data.find('MediaSettings').attrib.copy()
         self.policy = data.find('Policy').attrib.copy()
         self.location = data.find('Location').attrib.copy()
 
     def server(self):
-        server = list(filter(lambda x: x.machineIdentifier == self.machineIdentifier, self._servers))
+        server = list(filter(lambda x: x.clientIdentifier == self.machineIdentifier, self._server.resources()))
         if 0 == len(server):
             raise NotFound('Unable to find server with uuid %s' % self.machineIdentifier)
         return server[0]
 
-    def getMedia(self):
+    def getMedia(self, timeout=None):
```

Comment:
What up with the timeout? Why not just change the plex.Timeout?

Reply:
Not all operations require 60s+ timeout, and wrapping `getMedia()` calls with changing timeout back-and-forth would look strange.
You think that just globally change `TIMEOUT` in my code would be more than enough and wouldn't cause any strange problems?

Reply:
> You think that just globally change TIMEOUT in my code would be more than enough and wouldn't cause any strange problems?

It shouldn't cause any problems. timeout is essentially just telling the requests module when to give up if not getting a response. It's really only useful to change to poor or slow network issues. Having different timeouts for different calls in to the API seems incorrect unless you have a really strong case for it.